gnpy/core/network.py

```diff
@@ -50,9 +51,21 @@ def network_from_json(json_data, equipment):
     for el_config in json_data['elements']:
         typ = el_config.pop('type')
         variety = el_config.pop('type_variety', 'default')
-        if typ in equipment and variety in equipment[typ]:           
+        if typ in equipment and variety in equipment[typ]:
             extra_params = equipment[typ][variety]
-            el_config.setdefault('params', {}).update(extra_params.__dict__)
+            temp = el_config.setdefault('params', {})
+            # update native function ovrwrite existing values: use a home made function
```

Comment:
I've removed this comment. The function that we're calling is now named in a way which explains what this is about.